### PR TITLE
UCS/SYS/UCT: Avoid allocating huge pages if not supported by the system

### PR DESCRIPTION
**Pull request by yosefe**
enables running UCX on ppc64 VM

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7082/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7084/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
failure should be fixed by #2628

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7099/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7105/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@hoopoepg pls take a look

File: src/uct/base/uct_mem.c [hunk 1]
```diff
@@ -137,22 +138,26 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
 
         case UCT_ALLOC_METHOD_THP:
 #ifdef MADV_HUGEPAGE
+            /* Fixed option is not supported for thp allocation*/
```
Comment (hoopoepg) [inline src/uct/base/uct_mem.c]: minor: s/thp/the

Comment (yosefe) [reply src/uct/base/uct_mem.c]: here, thp is acronym for "transparent huge pages"

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7109/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7116/ for details (Mellanox internal link).

**Pull request merged**